integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md

```diff
@@ -0,0 +1,284 @@
+-u "${ES_USER}:${ES_PASSWORD}" \
+-H 'Content-Type: application/json' \
+-d'
+{
+  "properties": {
+    "passages": {
+      "type": "nested",
+      "properties": {
+        "sparse.tokens": {
+          "type": "rank_features"
+        }
+      }
+    }
+  }
+}'
+```
+The above command will update the index mappings to specify `passages` to be `nested` type and `passages.sparse.tokens` to be `rank_features` type, 
+because the default dynamic index mappings of the web crawler couldn't recognize these types correctly during document ingestion.  
+
+
+NOTE: `rank_features` only works for ELSER v1 model, ELSER v2 requires a different type for the tokens and ELSER v2 has only been available since Elastic 8.11. Learn more about ELSER v2 from [here](https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-elser.html)
+
+### Build a custom ingest pipeline with two processors
+Now you can build a custom ingest pipeline for your web crawler index on Kibana, following these steps:
+
+* Open http://localhost:5601, and log into Kibana with your Elasticsearch credentials, then go to your web crawler index page.
+
+
+* Under `Pipelines` tab, click on `Copy and customize` to create a custom ingest pipeline, and you will see a new ingest pipeline named `<your-index-name>@custom`.  
+  For example,  
+  <img src="assets/web_crawler_ingest_pipeline_custom.png" width="955" height="340" />
+
+
+* Click on `edit pipline` and then `Manage` -> `Edit`, it will take you to the ingest pipeline `Edit` page 
+where you can add processors to the pipeline.
+
+
+* Add a `Script` processor for chunking  
+  In the ingest pipeline page, click on `Add a processor`, choose `Script` processor, and then add a Groovy script to the `Source` field.  
+  For example,  
+  <img src="assets/web_crawler_script_processor.png" width="577" height="718" />
+
+  ```Groovy
+  String[] envSplit = /((?<!M(r|s|rs)\.)(?<=\.) |(?<=\!) |(?<=\?) )/.split(ctx['body_content']);
+  ctx['passages'] = new ArrayList();
+  int i = 0;
+  boolean remaining = true;
+  if (envSplit.length == 0) {
+      return
+  } else if (envSplit.length == 1) {
+      Map passage = ['text': envSplit[0]];ctx['passages'].add(passage)
+  } else {
+      while (remaining) {
+          Map passage = ['text': envSplit[i++]]
+          while (i < envSplit.length && passage.text.length() + envSplit[i].length() < params.model_limit) {passage.text = passage.text + ' ' + envSplit[i++]}
+          if (i == envSplit.length) {remaining = false}
+          ctx['passages'].add(passage)
+          }
+  }
+  ```
+  The above script splits `body_content` into sentences using regex, and then combine sentences into passages with 
+  a `model_limit` parameter to control the character length of each passage. `model_limit` is configured in `Parameters` field, for example,  
+  ```json
+  {
+    "model_limit": 2048
+  }
+  ```
+  
+  You can update the `Script` to build a customized chunking processor for your web content.  
```

Comment:
This section about updating the Script to build a customized chunking processor seems like a section that should be marked in a special way.  Like with a sub-heading indicating it is optional for consideration.   Up until this point in the document I was just following instructions.  But now I have to stop and read about something that is more like considerations